src/Storages/StorageFile.cpp

```diff
@@ -121,7 +121,7 @@ void listFilesWithFoldedRegexpMatchingImpl(const std::string & path_for_ls,
         return;
 
     const fs::directory_iterator end;
-    for (fs::directory_iterator it(path_for_ls); it != end; ++it)
+    for (fs::directory_iterator it(path_for_ls, fs::directory_options::skip_permission_denied); it != end; ++it)
```

Comment:
@zvonand : In the example, `SELECT *, _path, _file FROM file('{data1/f1,data2/f2}.csv', CSV)` if user doesn't have rights to `data2` looks like we would just skip it without throwing any error.

Reply:
@SmitaRKulkarni 
IMO we could append the docs with something like `a glob with '/' in curly braces will ignore directories that cannot be accessed`. 
I don't think that making the code twice as unreadable is a good idea for such a small issue.
Also, it doesn't affect _old_ workflow, it is _only_ about `{foo/bar,x/y/z}` globs.